velox/common/base/PeriodicStatsReporter.cpp

```diff
@@ -87,4 +101,177 @@ void PeriodicStatsReporter::reportArbitratorStats() {
       stats.freeReservedCapacityBytes);
 }
 
+void PeriodicStatsReporter::reportAllocatorStats() {
+  if (allocator_ == nullptr) {
+    return;
+  }
+  RECORD_METRIC_VALUE(
+      kMetricMappedMemoryBytes,
+      (velox::memory::AllocationTraits::pageBytes(allocator_->numMapped())));
+  RECORD_METRIC_VALUE(
+      kMetricAllocatedMemoryBytes,
+      (velox::memory::AllocationTraits::pageBytes(allocator_->numAllocated())));
+  // TODO(jtan6): Remove condition after T150019700 is done
+  if (auto* mmapAllocator =
+          dynamic_cast<const velox::memory::MmapAllocator*>(allocator_)) {
+    RECORD_METRIC_VALUE(
+        kMetricMmapRawAllocBytesSmall, (mmapAllocator->numMallocBytes()));
+    RECORD_METRIC_VALUE(
+        kMetricMmapExternalMappedBytes,
+        velox::memory::AllocationTraits::pageBytes(
+            (mmapAllocator->numExternalMapped())));
+  }
+  // TODO(xiaoxmeng): add memory allocation size stats.
+}
+
+void PeriodicStatsReporter::reportCacheStats() {
+  if (cache_ == nullptr) {
+    return;
+  }
+  const auto cacheStats = cache_->refreshStats();
+
+  // Snapshots.
+  RECORD_METRIC_VALUE(kMetricMemoryCacheNumEntries, cacheStats.numEntries);
```

Comment:
Since this is a velox change and you update SSD cache production with these non-average stats? Or you first report average stats and does the non-average stats in followup? Thanks!